embassy-stm32/src/tsc/mod.rs

```diff
@@ -90,6 +93,35 @@ pub enum Error {
     Test,
 }
 
+/// Async acquisition API marker
+pub struct Async;
+/// Blocking acquisition API marker
+pub struct Blocking;
+
+trait SealedDriverKind {}
+
+impl SealedDriverKind for Async {}
+impl SealedDriverKind for Blocking {}
+
+#[allow(private_bounds)]
+/// Driver variant marker for the TSC peripheral
+pub trait DriverKind: SealedDriverKind {}
+
+impl DriverKind for Async {}
+impl DriverKind for Blocking {}
```

Comment:
Instead of defining your own, use the ones from https://github.com/embassy-rs/embassy/blob/main/embassy-stm32/src/lib.rs#L21-L41  (see SPI for example).

Reply:
Thank you, changed!